src/app/data-explorer/data-schema-row/data-schema-row.component.ts

```diff
@@ -0,0 +1,171 @@
+/*
+Copyright 2022-2023 University of Oxford
+and Health and Social Care Information Centre, also known as NHS Digital
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+import {
+  Component,
+  EventEmitter,
+  Input,
+  OnChanges,
+  OnInit,
+  Output,
+  SimpleChanges,
+} from '@angular/core';
+import { MatCheckboxChange } from '@angular/material/checkbox';
+import { RequestElementAddDeleteEvent } from 'src/app/shared/data-element-in-request/data-element-in-request.component';
+import {
+  DataClassWithElements,
+  DataElementSearchResult,
+  DataItemDeleteEvent,
+  DataRequestQueryType,
+  DataSchema,
+  // RefreshRequestEvent,
+  SelectionChange,
+  SelectionChangedBy,
+} from '../data-explorer.types';
+import { DataAccessRequestsSourceTargetIntersections } from '../data-requests.service';
+import { DataSchemaService } from 'src/app/mauro/data-schema-service';
+
+@Component({
+  selector: 'mdm-data-schema-row',
+  templateUrl: './data-schema-row.component.html',
+  styleUrls: ['./data-schema-row.component.scss'],
+})
+export class DataSchemaRowComponent implements OnInit, OnChanges {
+  @Input() dataSchema?: DataSchema;
+  @Input() requestName = '';
+  @Input() requestId = '';
+  @Input() suppressViewRequestsDialogButton = false;
+  @Input() canDelete = true;
+  @Input() sourceTargetIntersections: DataAccessRequestsSourceTargetIntersections;
+  @Input() allSelected?: SelectionChange;
+
+  @Output() deleteItemEvent = new EventEmitter<DataItemDeleteEvent>();
+  @Output() requestAddDelete = new EventEmitter<RequestElementAddDeleteEvent>();
+  // @Output() requestCreated = new EventEmitter<CreateRequestEvent>();
+  @Output() schemaCheckedEvent = new EventEmitter();
+  @Output() setRemoveSelectedButtonDisabledEvent = new EventEmitter();
+
+  visible = true;
+
+  schemaSelected: SelectionChange = {
+    changedBy: { instigator: 'parent' },
+    isSelected: false,
+  };
+  schemaElements: DataElementSearchResult[] = [];
+
+  cohortQueryType: DataRequestQueryType = 'cohort';
+  dataQueryType: DataRequestQueryType = 'data';
+
+  constructor(private dataSchemaService: DataSchemaService) {
+    this.sourceTargetIntersections = {
+      dataAccessRequests: [],
+      sourceTargetIntersections: [],
+    };
+  }
+
+  // Init, get the classes from the request. and list.
```

Comment:
I feel this comment is pointing out the obvious a bit. Another way to make the function clearer is to rename the service method to `this.dataSchemaService.getDataSchemaElements()`

Reply:
The comment is wrong. The line of code that follows it doesn't even do what the comments says. Comment deleted and service methods have been renamed to be prefixed with "get" to make things clearer as suggested.